rust/src/ketama.rs

```diff
@@ -40,38 +77,82 @@ impl KetamaPool {
         };
         self.ranking[ranking_idx % self.ranking.len()].index as usize
     }
-}
 
-const POINTS_PER_HASH: usize = 4;
-const POINTS_PER_SERVER: usize = 40;
+    fn update_node_ranking(&mut self) {
+        self.ranking.clear();
```

Comment:
Doesn't this mean you have to recreate the entire ranking on every addition and deletion? Maybe I'm misunderstanding something, but I thought the point of consistent hashing was not to do this.

Reply:
I added the comment noting that we can incrementally `add` if we really wanted to, but its probably not worth it.